xds/src/test/java/io/grpc/xds/internal/GoogleDefaultXdsCredentialsProviderTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.internal;
+
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+import io.grpc.CompositeChannelCredentials;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link GoogleDefaultXdsCredentialsProvider}. */
+@RunWith(JUnit4.class)
+public class GoogleDefaultXdsCredentialsProviderTest {
```

Comment:
For each of the providers, also have a test that checks the provider is available from the service loader. See https://github.com/grpc/grpc-java/blob/cda0e9d996574bb392d9fc06562b8b3931893d95/netty/src/test/java/io/grpc/netty/NettyChannelProviderTest.java#L40

That would detect issues in the xds/src/main/resources/META-INF/services/io.grpc.xds.XdsCredentialsProvider file.

Reply:
Done